include/cadmium/engine/pdevs_simulator.hpp

```diff
@@ -216,20 +163,124 @@ namespace cadmium {
                     }
                 }
 
-                auto log_state = [](const typename model_type::state_type& s) -> std::string {
-                    std::ostringstream oss;
-                    oss << "State for model ";
-                    oss << boost::typeindex::type_id<model_type>().pretty_name();
-                    oss << " is ";
-                    oss << s;
-                    return oss.str();
-                };
-
-                LOGGER::template log<cadmium::logger::logger_state, decltype(log_state), const typename model_type::state_type&>(log_state, _model.state);
+                LOGGER::template log<cadmium::logger::logger_state, decltype(formatter::log_state), const typename model_type::state_type&>(formatter::log_state, _model.state);
             }
     //TODO: use enable_if functions to give access to read state and messages in debug mode
         };
 
+
+        template<template<typename T> class MODEL, typename TIME, typename LOGGER>
+        class dynamic_simulator {
+            using formatter=typename cadmium::logger::simulator_formatter<MODEL, TIME>;
+
+            using input_ports=typename modeling::dynamic_atomic<MODEL, TIME>::input_ports;
+            using output_ports=typename modeling::dynamic_atomic<MODEL, TIME>::output_ports;
+            using in_bags_type=typename make_message_bags<input_ports>::type;
+            using out_bags_type=typename make_message_bags<output_ports>::type;
+            modeling::dynamic_atomic<MODEL, TIME> _model;
+            TIME _last;
+            TIME _next;
+
+            dynamic_message_bags _outbox;
+            dynamic_message_bags _inbox;
+
+        public:
+            using model_type=MODEL<TIME>;
+
+            /**
+             * @brief constructor is used as init function, sets the start time
```

Comment:
Constructor is not used as init function here.

Reply:
This comment was dragged from the original simulator class comment. I have modified it now.
Done.